uwblueprint/supportive-housing
# Fix Logout When Token Expires

## Notion task link


[Fix Logout When Token Expires](https://www.notion.so/uwblueprintexecs/Fix-Logout-When-Token-Expires-fa12cf98161e486699be3c796264a5fb?pvs=4)



## Implementation description
- This was a documented issue with starter code V2 [here](https://www.notion.so/uwblueprintexecs/Handling-refresh-tokens-INVALID_REFRESH_TOKEN-issue-2e69d2010e1949569c0be2f75f22e130)
- When we had a valid refresh token and needed to set the new access token, the `Bearer` keyword was missing in the header
- When the refresh token was invalid, the user was not forced to logout and get a new token
- The refresh token lifespan was also changed to be 24 hours to limit the number of sign-ins needed per user (this is also the standard for single page applications



## Steps to test
Testing that a missing refresh token is handled:

- Log in to your local environment.
- Set your system time to a time at least a week in the future.
- Close your browser fully and reopen it.
- Load any page that pulls data from the backend, and verify that you are logged out.



## What should reviewers focus on?
* 


## Checklist
